### PR TITLE
Remove fetch all databases

### PR DESCRIPTION
**Pull request by parth-crest**
## Closes https://github.com/elastic/connectors-py/issues/171

Updated code to index nothing if user has passed empty value in "List of MySQL databases" from the UI.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally

### CODE REVIEW

**Comment by parth-crest** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
buildkite test this

File: connectors/sources/mysql.py [hunk 1]
```diff
@@ -331,37 +331,24 @@ async def fetch_documents(self, database, table, query):
...
-
     async def get_docs(self):
         """Executes the logic to fetch databases, tables and rows in async manner.
 
         Yields:
             dictionary: Row dictionary containing meta-data of the row.
         """
         database_config = self.configuration["database"]
-        if isinstance(database_config, str):
+        if not database_config:
```
Comment (tarekziade) [inline connectors/sources/mysql.py:341]: let's be explicit.

`if  database_config == ''`

File: connectors/sources/mysql.py [hunk 2]
```diff
@@ -331,37 +331,24 @@ async def fetch_documents(self, database, table, query):
...
+            logger.info(
+                "No database passed in the configuration. Stopping the iteration."
+            )
+            return
+        elif isinstance(database_config, str):
```
Comment (tarekziade) [inline connectors/sources/mysql.py:346]: can it be anything else than `str` ? if it's not `str` what do we do?

**Review by tarekziade — CHANGES_REQUESTED** [review]
I know there's another patch for field validation, but here the logic is not great because the `if` test is not explicit. Maybe treat the empty string, the string and then raise of you get something else? (until we have the other patch) (or maybe this patch can be combined with the other one)

**Comment by parth-crest** [comment]
Hey @tarekziade , previously we were explicitly calling the if condition as we were getting empty list ([]) because we were using get_default_configuration() method to get config. So we were checking on `len(database)`, However our framework is evolving and with the newer update we are getting empty string('') from the UI, that change broke the logic. So we updated if condition to be implicit to cover more cases.
But from now onwards we will take care to handle if condition explicitly.

Also, as you suggested we can close this PR and accommodate the changes in the Validation Config PR.

**Comment by tarekziade** [comment]
> Hey @tarekziade , previously we were explicitly calling the if condition as we were getting empty list ([]) because we were using get_default_configuration() method to get config. So we were checking on `len(database)`, However our framework is evolving and with the newer update we are getting empty string('') from the UI, that change broke the logic.

yes, and the changed logic is fine

> So we updated if condition to be implicit to cover more cases. But from now onwards we will take care to handle if condition explicitly.

yes thanks, the problem with implicit if is that your value is converted to a boolean. and that leads to unexpected results.
like, do you know what `bool(-1)` or `bool(0)` returns?  ;)

for this reason, since Python is not statically typed, it's best to be explicit and not leave room for uncertainty.


> 
> Also, as you suggested we can close this PR and accommodate the changes in the Validation Config PR.

sounds good

File: connectors/sources/mysql.py (hunk 1 above)
Comment (parth-crest) [reply connectors/sources/mysql.py:341]: Sure we will update that in the other Validate PR.

File: connectors/sources/mysql.py (hunk 2 above)
Comment (parth-crest) [reply connectors/sources/mysql.py:346]: We can have database_config from either UI or from the get_default_configuration() method.

- From the UI we will be getting string everytime.
- From the get default configuration method we will have list and we have handled that case in the else condition.

**Pull request closed without merging**